File: Python/Coffee Machine - PYTHON/Coffee Machine/task/stage_2/coffee_machine.py
class CoffeeMachine:
    INGREDIENTS_PER_CUP = {'water': 200, 'milk': 50, 'beans': 15}

    @staticmethod
    def calculate_cups(cups: int) -> tuple:
        water = CoffeeMachine.INGREDIENTS_PER_CUP['water'] * cups
        milk = CoffeeMachine.INGREDIENTS_PER_CUP['milk'] * cups
        beans = CoffeeMachine.INGREDIENTS_PER_CUP['beans'] * cups
        return water, milk, beans


class UserInterface:

    def main(self) -> None:
        print("Write how many cups of coffee you will need:")
        cups_needed = self._validate_input(input())

        if cups_needed:
            self._display_amounts(cups_needed)
        else:
            self.main()

    @staticmethod
    def _display_amounts(cups_needed: list) -> None:
        water, milk, beans = CoffeeMachine.calculate_cups(cups_needed)
        print(f"For {cups_needed} cups of coffee you will need:")
        print(f"{water} ml of water")
        print(f"{milk} ml of milk")
        print(f"{beans} g of coffee beans")

    @staticmethod
    def _validate_input(user_input: str) -> int:
        try:
            num_input = int(user_input)
            return num_input
        except ValueError:
            print("Please provide an integer")
            return None


if __name__ == '__main__':
    interface = UserInterface()
    interface.main()

File: Java/Coffee Machine/step2/CoffeeMachine.java
package machine;

import java.util.Scanner;

public class CoffeeMachine {
    private static final Scanner scanner = new Scanner(System.in);

    public static void main(String[] args) {
        System.out.println("Write how many cups of coffee you will need:");
        int cups = scanner.nextInt();

        System.out.print("For " + cups + (cups == 1 ? " cup " : " cups ") + "of coffee you will need:");
        System.out.println(calculateIngredient(cups, "water") + " ml of water");
        System.out.println(calculateIngredient(cups, "milk") + " ml of milk");
        System.out.println(calculateIngredient(cups, "beans") + " g of coffee beans");


    }

    public static int calculateIngredient(int cups, String ingredient) {
        int returnedMeasurement = 0;
        switch (ingredient) {
            case "water":
                returnedMeasurement = cups * 200;
                break;
            case "milk":
                returnedMeasurement = cups * 50;
                break;
            case "beans":
                returnedMeasurement = cups * 15;
            default:
                break;
        }
        return returnedMeasurement;
    }
}
